golang/frontend/main.go

```diff
@@ -1,104 +1,141 @@
-	messageServiceUrl = os.Getenv("MESSAGE_ENDPOINT") + "/message"
-)
-
-func main() {
-	exp, err := honeycomb.NewExporter(
-		honeycomb.Config{
-			APIKey: apiKey,
-		},
-		honeycomb.TargetingDataset(dataset),
-		honeycomb.WithServiceName("frontend-go"),
-	)
-	if err != nil {
-		log.Fatal(err)
-	}
-	defer exp.Close()
-
-	config := sdktrace.Config{
-		DefaultSampler: sdktrace.AlwaysSample(),
-	}
-	tp, err := sdktrace.NewProvider(sdktrace.WithConfig(config), sdktrace.WithSyncer(exp))
-	if err != nil {
-		log.Fatal(err)
-	}
-	global.SetTraceProvider(tp)
-
-	tracer := global.Tracer("greeting-service/frontend")
-
-	mux := http.NewServeMux()
-	mux.HandleFunc("/greeting", func(w http.ResponseWriter, r *http.Request) {
-		attrs, entries, spanCtx := httptrace.Extract(r.Context(), r)
-
-		r = r.WithContext(correlation.ContextWithMap(r.Context(), correlation.NewMap(correlation.MapUpdate{MultiKV: entries})))
-
-		ctx, span := tracer.Start(
-			trace.ContextWithRemoteSpanContext(r.Context(), spanCtx), "/greeting",
-			trace.WithAttributes(attrs...),
-		)
-		defer span.End()
-
-		name := getName(ctx)
-		message := getMessage(ctx)
-
-		fmt.Fprintf(w, "Hello %s, %s", name, message)
-	})
-
-	log.Fatal(http.ListenAndServe(":7000", mux))
-}
-
-func getName(ctx context.Context) string {
-	tracer := global.Tracer("greeting-service/frontend")
-	var getNameSpan trace.Span
-	ctx, getNameSpan = tracer.Start(ctx, "✨ call /name ✨")
-	defer getNameSpan.End()
-	return makeRequest(ctx, nameServiceUrl)
-}
-
-func getMessage(ctx context.Context) string {
-	tracer := global.Tracer("greeting-service/frontend")
-	var getMessageSpan trace.Span
-	ctx, getMessageSpan = tracer.Start(ctx, "✨ call /message ✨")
-	defer getMessageSpan.End()
-	return makeRequest(ctx, messageServiceUrl)
-}
-
-func makeRequest(ctx context.Context, url string) string {
-	req, _ := http.NewRequest("GET", url, nil)
-	ctx, req = httptrace.W3C(ctx, req)
-	httptrace.Inject(ctx, req)
-	client := http.DefaultClient
-	res, err := client.Do(req)
-	if err != nil {
-		panic(err)
-	}
-	body, err := ioutil.ReadAll(res.Body)
-	res.Body.Close()
-	if err != nil {
-		panic(err)
-	}
-	return string(body)
-}
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	nameServiceUrl    = os.Getenv("NAME_ENDPOINT") + "/name"
+	messageServiceUrl = os.Getenv("MESSAGE_ENDPOINT") + "/message"
+	tracer            trace.Tracer
+)
+
+func getGrpcEndpoint() string {

```

Comment:
attempting to let OTEL and beeline to use the same env var. unfortunately, both of them are very sensitive about the URL shape